(Nightly) Builds fail against netgen 6.2.2601

### Problem description

Nightly builds for F43/F44/Rawhide started failing recently, ever since @manisandro bumped netgen-mesher to 6.2.2601.

I have an active copr nightly builds, the only one currently succeeding is the aarch64 on F43, but only because netgen-mesher is still 6.2.2505.

Failure seems to be because of undefined reference to 'netgen::multithread' . 6.2.2601 .so files do not have that symbol any more. I'm attaching the build log file. 

[builder-live.log](https://github.com/user-attachments/files/25172367/builder-live.log)

### Workbench affected?

FEM

### Steps to reproduce

Try to build FreeCAD from the git source against netgen-mesher 6.2.2601

### Expected behavior

FreeCAD builds against the newer version of netgen-mesher

### Actual behavior

FreeCAD fails to build against netgen-mesher 2.6.2601

### Development version About Info (in Safe Mode)

```shell
n/a
```

### Last known good version (optional)

```shell

```

i saw this earlier today and it garnered my interest. so i ended setting up a homebrew formula for this version of netgen you mention above ie. 6.2.2601 and attempted to build freecad with fem netgen support ie. using,

```
echo $pthng
# /home/capin/homebrew/opt/netgen@6.2.2601
```

```
# omitted all other cmake vars for posterity

cmake -DBUILD_FEM_NETGEN=ON \
-DCMAKE_PREFIX_PATH=$pthng; \
../../fcsrc 
```

and sure enough i'm seeing the below error,

```
[2160/6963] Linking CXX shared library lib/libNETGENPlugin.so
FAILED: [code=1] lib/libNETGENPlugin.so
: && /home/capin/homebrew/bin/clang++ -fPIC -Wall -Wextra -Wpedantic -Wno-write-strings -fdiagnostics-color -I/home/capin/homebrew/opt/mesa/include -I/home/capin/homebrew/opt/pybind11_py312/include -I/home/capin/homebrew/opt/mesa-glu/include -D_GLIBCXX_ASSERTIONS -Wno-undefined-var-template -Wno-self-assign -Wno-sign-compare -Wno-logical-op-parentheses -Wno-reorder -Wno-switch -Wno-switch-enum -Wno-unknown-pragmas -Wno-unused-variable -Wno-unused-private-field -Wno-unused-function -Wno-sometimes-uninitialized -Wno-overloaded-virtual -Wno-dynamic-class-memaccess -Wno-comment -Wno-unused-parameter -Wno-extra-semi -Wno-deprecated-copy -Wno-missing-field-initializers -Wno-c++17-extensions -O2 -g -DNDEBUG  -shared  -Wl,--no-undefined -Xlinker --dependency-file=src/3rdParty/salomesmesh/CMakeFiles/NETGENPlugin.dir/link.d -Wl,-soname,libNETGENPlugin.so -o lib/libNETGENPlugin.so src/3rdParty/salomesmesh/CMakeFiles/NETGENPlugin.dir/src/NETGENPlugin/NETGENPlugin_Hypothesis.cpp.o src/3rdParty/salomesmesh/CMakeFiles/NETGENPlugin.dir/src/NETGENPlugin/NETGENPlugin_Hypothesis_2D.cpp.o src/3rdParty/salomesmesh/CMakeFiles/NETGENPlugin.dir/src/NETGENPlugin/NETGENPlugin_Mesher.cpp.o src/3rdParty/salomesmesh/CMakeFiles/NETGENPlugin.dir/src/NETGENPlugin/NETGENPlugin_NETGEN_2D.cpp.o src/3rdParty/salomesmesh/CMakeFiles/NETGENPlugin.dir/src/NETGENPlugin/NETGENPlugin_NETGEN_2D3D.cpp.o src/3rdParty/salomesmesh/CMakeFiles/NETGENPlugin.dir/src/NETGENPlugin/NETGENPlugin_NETGEN_2D_ONLY.cpp.o src/3rdParty/salomesmesh/CMakeFiles/NETGENPlugin.dir/src/NETGENPlugin/NETGENPlugin_NETGEN_3D.cpp.o src/3rdParty/salomesmesh/CMakeFiles/NETGENPlugin.dir/src/NETGENPlugin/NETGENPlugin_SimpleHypothesis_2D.cpp.o src/3rdParty/salomesmesh/CMakeFiles/NETGENPlugin.dir/src/NETGENPlugin/NETGENPlugin_SimpleHypothesis_3D.cpp.o -L/home/capin/homebrew/opt/opencascade/lib -Wl,-rpath,/home/capin/homebrew/opt/opencascade/lib:/opt/code/fcgit/builds/issue.tshooting.qt6.py313/lib:/home/capin/homebrew/opt/zlib/lib:/home/capin/homebrew/opt/libaec/lib:/home/capin/homebrew/opt/vtk@9.5.2_py313/lib:/home/capin/homebrew/opt/hdf5/lib:/home/capin/homebrew/opt/freetype/lib:/home/capin/homebrew/opt/libx11/lib:/home/capin/homebrew/Cellar/boost/1.90.0_1/lib:/home/capin/homebrew/opt/med-file/lib:/home/capin/homebrew/opt/netgen@6.2.2601/lib::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::  lib/libStdMeshers.so  lib/libSMESH.so  lib/libDriverSTL.so  lib/libDriverDAT.so  lib/libDriverUNV.so  lib/libSMESHDS.so  lib/libSMDS.so  /home/capin/homebrew/opt/hdf5/lib/libhdf5.a  -lm  /home/capin/homebrew/opt/zlib/lib/libz.so  /home/capin/homebrew/opt/libaec/lib/libsz.so.2.0.1  /home/capin/homebrew/opt/libaec/lib/libaec.so.0.1.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkWrappingPythonCore3.13-9.5.so.9.5  /home/capin/homebrew/opt/hdf5/lib/libhdf5.so.320.0.0  /home/capin/homebrew/opt/hdf5/lib/libhdf5_hl.so.320.0.0  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkInteractionStyle-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkFiltersExtraction-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkIOXML-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkIOXMLParser-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkIOLegacy-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkIOCellGrid-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkIOCore-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkFiltersCellGrid-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkRenderingOpenGL2-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkRenderingFreeType-9.5.so.9.5  /home/capin/homebrew/opt/freetype/lib/libfreetype.so  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkIOImage-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkRenderingHyperTreeGrid-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkRenderingUI-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkRenderingCore-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkFiltersSources-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkFiltersGeneral-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkFiltersVerdict-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkFiltersGeometry-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkFiltersCore-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkglad-9.5.so.9.5  /home/capin/homebrew/opt/libx11/lib/libX11.so  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkImagingCore-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkCommonExecutionModel-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkCommonDataModel-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkCommonTransforms-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkCommonMisc-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkCommonMath-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkkissfft-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtkCommonCore-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtktoken-9.5.so.9.5  /home/capin/homebrew/opt/vtk@9.5.2_py313/lib/libvtksys-9.5.so.9.5  /home/capin/homebrew/Cellar/boost/1.90.0_1/lib/libboost_program_options.so.1.90.0  /home/capin/homebrew/Cellar/boost/1.90.0_1/lib/libboost_regex.so.1.90.0  /home/capin/homebrew/Cellar/boost/1.90.0_1/lib/libboost_thread.so.1.90.0  /home/capin/homebrew/Cellar/boost/1.90.0_1/lib/libboost_container.so.1.90.0  /home/capin/homebrew/Cellar/boost/1.90.0_1/lib/libboost_atomic.so.1.90.0  /home/capin/homebrew/Cellar/boost/1.90.0_1/lib/libboost_chrono.so.1.90.0  /home/capin/homebrew/Cellar/boost/1.90.0_1/lib/libboost_date_time.so.1.90.0  -lpthread  lib/libDriver.so  /home/capin/homebrew/opt/med-file/lib/libmedC.so  /home/capin/homebrew/opt/med-file/lib/libmed.so  lib/libMEFISTO2.so  /home/capin/homebrew/opt/netgen@6.2.2601/lib/libnglib.so  /home/capin/homebrew/opt/netgen@6.2.2601/lib/libngcore.so  -ldl  /home/capin/homebrew/opt/opencascade/lib/libTKOffset.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKFillet.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKFeat.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKExpress.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKDESTEP.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKDEIGES.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKXSBase.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKBool.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKDEGLTF.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKDESTL.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKBin.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKBinL.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKMeshVS.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKRWMesh.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKXCAF.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKVCAF.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKCAF.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKBO.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKPrim.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKLCAF.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKCDF.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKV3d.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKMesh.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKShHealing.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKHLR.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKTopAlgo.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKGeomAlgo.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKService.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKDE.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKBRep.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKGeomBase.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKG3d.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKG2d.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKMath.so.7.9.3  /home/capin/homebrew/opt/opencascade/lib/libTKernel.so.7.9.3  -Wl,-rpath-link,/home/capin/homebrew/opt/vtk@9.5.2_py313/lib && :
/usr/bin/ld: src/3rdParty/salomesmesh/CMakeFiles/NETGENPlugin.dir/src/NETGENPlugin/NETGENPlugin_Mesher.cpp.o: in function `NETGENPlugin_Mesher::Compute()':
/opt/code/fcgit/fcsrc/src/3rdParty/salomesmesh/src/NETGENPlugin/NETGENPlugin_Mesher.cpp:2576:(.text+0xa8ac): undefined reference to `netgen::multithread'
/usr/bin/ld: /opt/code/fcgit/fcsrc/src/3rdParty/salomesmesh/src/NETGENPlugin/NETGENPlugin_Mesher.cpp:2576:(.text+0xa8b0): undefined reference to `netgen::multithread'
/usr/bin/ld: /opt/code/fcgit/fcsrc/src/3rdParty/salomesmesh/src/NETGENPlugin/NETGENPlugin_Mesher.cpp:2676:(.text+0xaf80): undefined reference to `netgen::multithread'
/usr/bin/ld: /opt/code/fcgit/fcsrc/src/3rdParty/salomesmesh/src/NETGENPlugin/NETGENPlugin_Mesher.cpp:2676:(.text+0xaf84): undefined reference to `netgen::multithread'
/usr/bin/ld: /opt/code/fcgit/fcsrc/src/3rdParty/salomesmesh/src/NETGENPlugin/NETGENPlugin_Mesher.cpp:2755:(.text+0xb304): undefined reference to `netgen::multithread'
/usr/bin/ld: src/3rdParty/salomesmesh/CMakeFiles/NETGENPlugin.dir/src/NETGENPlugin/NETGENPlugin_Mesher.cpp.o:/opt/code/fcgit/fcsrc/src/3rdParty/salomesmesh/src/NETGENPlugin/NETGENPlugin_Mesher.cpp:2755: more undefined references to `netgen::multithread' follow
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
[2169/6963] Building CXX object tests/lib/googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.o
In file included from /opt/code/fcgit/fcsrc/tests/lib/googletest/src/gtest-all.cc:38:
In file included from /opt/code/fcgit/fcsrc/tests/lib/googletest/include/gtest/gtest.h:65:
In file included from /opt/code/fcgit/fcsrc/tests/lib/googletest/include/gtest/gtest-death-test.h:43:
In file included from /opt/code/fcgit/fcsrc/tests/lib/googletest/include/gtest/internal/gtest-death-test-internal.h:47:
In file included from /opt/code/fcgit/fcsrc/tests/lib/googletest/include/gtest/gtest-matchers.h:49:
/opt/code/fcgit/fcsrc/tests/lib/googletest/include/gtest/gtest-printers.h:502:35: warning: implicit conversion from 'char8_t' to 'char32_t' may change the meaning of the represented code unit [-Wcharacter-conversion]
  502 |   PrintTo(ImplicitCast_<char32_t>(c), os);
      |           ~~~~~~~~~~~~~           ^
1 warning generated.
```

--- a/src/3rdParty/salomesmesh/src/NETGENPlugin/NETGENPlugin_Mesher.cpp
+++ b/src/3rdParty/salomesmesh/src/NETGENPlugin/NETGENPlugin_Mesher.cpp
@@ -126,7 +126,11 @@ namespace netgen {
 #endif
   //extern void OCCSetLocalMeshSize(OCCGeometry & geom, Mesh & mesh);
   DLL_HEADER extern MeshingParameters mparam;
-  DLL_HEADER extern volatile multithreadt multithread;
+#if NETGEN_VERSION >= NETGEN_VERSION_STRING(6,2,2601)
+    using ngcore::multithread;
+#else
+    DLL_HEADER extern volatile multithreadt multithread;
+#endif
   DLL_HEADER extern bool merge_solids;
 }
 

--- a/src/3rdParty/salomesmesh/src/NETGENPlugin/NETGENPlugin_NETGEN_3D.cpp
+++ b/src/3rdParty/salomesmesh/src/NETGENPlugin/NETGENPlugin_NETGEN_3D.cpp
@@ -112,7 +112,11 @@ namespace netgen {
   DLL_HEADER extern int OCCGenerateMesh (OCCGeometry&, Mesh*&, int, int, char*);
 #endif
   DLL_HEADER extern MeshingParameters mparam;
-  DLL_HEADER extern volatile multithreadt multithread;
+#if NETGEN_VERSION >= NETGEN_VERSION_STRING(6,2,2601)
+    using ngcore::multithread;
+#else
+    DLL_HEADER extern volatile multithreadt multithread;
+#endif
 }
 using namespace nglib;
 using namespace std;

